Nakama/HttpRequestAdapter.cs

```diff
@@ -142,7 +142,19 @@ public static IHttpAdapter WithGzip(bool decompression = false, bool compression
 
         private bool IsTransientException(Exception e)
         {
-            return (e is ApiResponseException apiException && (apiException.StatusCode >= 500 || apiException.StatusCode == -1)) || e is HttpRequestException;
+            if (e is ApiResponseException apiException)
+            {
+                switch (apiException.StatusCode)
+                {
+                    case 500:
+                    case 502:
+                    case 503:
+                    case 504:
+                        return true;
+                }
+            }
+
+            return false;
```

Comment:
Are we intentionally removing `-1` from the list of monitored exception status codes here?
Also, I would tend to agree that we probably don't want to retry on an Internal Server Error as it's likely that more often than not the same error will occur if nothing about the request has changed.